cal/cal/static/js/app.js

```diff
@@ -1,111 +1,124 @@
+var analyticsApp = window.angular.module('analyticsApp', ['analyticsApp.services', 'nvd3', 'ui.calendar']);
+
+analyticsApp.controller('LoggedInCtrl', function LoggedInController() {
 });
 
-function TagsCtrl($scope, $http) {
-  var tagUrl = '/v1/tags';
-  $scope.tags = [];
+function TagListCtrl($scope, $http, CalendarRangeService, TagService) {
 
-  // add all the tags
-  $http({method: 'GET', url: tagUrl + '.json' }).
-  success(function successCallback(data) {
-    for (var i = 0; i < data.results.length; i++) {
-      var tag = data.results[i];
-      $scope.tags.push({
-        id: tag.id,
-        label: tag.label,
-        keywords: tag.keywords,
-        hours: tag.hours
-      });
+  var _this = this;
+
+  this.tags = [];
+
+  $scope.$on('calendarRange:updated', function(event, data) {
+    /* jshint unused:vars */
+    var rangeData = CalendarRangeService.getRange();
+    var timeRange = rangeData.timeRange;
+    if (!_this.isCumulative) {
+      _this.getTags('cumulative', null, null)
+        .then(function(tags) {
+          _this.tags = tags;
+          _this.timeRange = timeRange;
+        });
     }
   });
 
+  // Initialization
+  this.initialize = function() {
+
+    var tagsPromise;
+
+    if (this.isCumulative) {
+      tagsPromise = TagService.getTags('cumulative', null, null);
+    } else {
+      var initialTimeRange = CalendarRangeService.getRange();
+      tagsPromise = TagService.getTags(initialTimeRange.timeRange, initialTimeRange.start, initialTimeRange.end);
+    }
+    tagsPromise.then(function(tags) {
+      _this.tags = tags;
+    });
+  }.bind(this);
+  this.initialize();
+
+  this.hideZeroHoursFilter = function (value, index, array) {
+    /* jshint unused:vars */
+    if (this.hideZeroHours && value.hours === 0) {
+      return false;
+    } else {
+      return true;
+    }
+  }.bind(this);
+
   this.create = function(tag) {
-    $http({
-      method: 'POST',
-      url: tagUrl + '.json',
-      data: $.param({
-        label: tag.label,
-        keywords: tag.keywords,
-        csrfmiddlewaretoken: getCookie('csrftoken')
-      }),
-      headers: {
-        'Content-Type': 'application/x-www-form-urlencoded'
-      }
-    }).
-    success(function addToList(data) {
-      $scope.tags.push({
-        id: data.id,
-        label: data.label,
-        keywords: data.keywords,
-        hours: data.hours,
-        editing: false
+    TagService.createTag(tag.label, tag.keywords)
+      .success(function addToList(data) {
+        _this.tags.push({
+          id: data.id,
+          label: data.label,
+          keywords: data.keywords,
+          hours: data.hours,
+          editing: false
+        });
       });
-    });
-  };
+  }.bind(this);
 
   this.startEdit = function(tagId) {
-    var tag = $scope.tags.find(function(tag, index, array) { return tag.id == tagId; });
+    var tag = this.tags.find(function(tag, index, array) {
+      /* jshint unused:vars */
```

Comment:
This will prevent your own linter from yelling at you when you install it. Otherwise all us will have to configure our `.jshintrc`'s, which imo isn't worth the effort